users/konstantin/rules.mk

```
@@ -1,12 +1,15 @@
 SRC += konstantin.c
-ifneq (,$(filter yes,$(RGBLIGHT_ENABLE) $(RGB_MATRIX_ENABLE)))  # if either is yes
-  SRC += rgb.c
+ifeq ($(strip $(RGBLIGHT_ENABLE)), yes)
+    SRC += rgb.c
+endif
+ifneq (,$(filter yes IS31FL3731 IS31FL3733,$(RGB_MATRIX_ENABLE)))
```

Might be easier to just add this if it's not set to "no". 

```suggestion
ifneq ($(strip $(RGB_MATRIX_ENABLE)), no)
```

As is, this mises the 3737, and WS2812.  I suspect you're not using them, but it will still miss them.

---

Thanks, will change that.

---

Will your suggestion work if `RGB_MATRIX_ENABLE` is undefined, i.e. if none of the `rules.mk` down the line assign it to to either `no` or one of the values?

---

`RGB_MATRIX_ENABLE` is never not defined, in theory.  It should be set to `no` by default, actually. 
https://github.com/qmk/qmk_firmware/blob/22ba36a4d86c06ce1b6049555525646eb16bb57a/common_features.mk#L139-L144

---

Hmm, it'd be cool if that were the case for most/all `*_ENABLE` flags. Then we wouldn't need to check if they're `yes` and handle special cases like `full` and `lite` for Bootmagic — instead, we could just uniformly check if they're not `no`.

---

Well, for a majority of the features, there is only "yes" that gets checked.  The RGB/LED Matrix is one of the few settings that actually has multiple values.  So, it's one of the ones that is set by default.

For the other settings, checking to see if it's set to "yes" will work just fine.

---

Yeah, but I think that having all checks be of the same form is a benefit unto itself. It reduces the chance of errors and is more future-proof. For example:

Say that you're checking if a flag is `yes` in your userspace `rules.mk`, as such: `ifeq ($(strip $(<feature>_ENABLE)), yes)`. Then say the feature gets changed to have different values (like Bootmagic was changed to have `full` and `lite` alongside `yes`). All of a sudden your source files aren't being included for boards that use one of the new values instead of `yes`.

If we preemptively add `?= no` defaults for all flags, you can write all the checks as `ifneq ($(strip $(<feature>_ENABLE)), no)`, and this is no longer an issue.